### PR TITLE
EN-171 Converter cannot download some files

### PR DESCRIPTION
**Pull request by kacperwleklak**

### CODE REVIEW

**Review by ptylczynski — APPROVED** [review]
seems good, tested and works

File: src/main/java/pl/psnc/dei/iiif/Converter.java [hunk 1]
```diff
@@ -201,8 +211,13 @@ private void replaceUrl(ConversionDataHolder.ConversionData data) throws Malform
 	 * @param file file to which save contents
 	 * @throws IOException
 	 */
-	private void copyURLToFile(URL url, File file) throws IOException {
-		HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+	private void copyURLToFile(URL url, File file) throws IOException, DownloadFileException {
+		HttpURLConnection connection;
+		try {
+			connection = (HttpURLConnection) url.openConnection();
+		} catch (ClassCastException e) {
```
Comment (helinek) [inline src/main/java/pl/psnc/dei/iiif/Converter.java:218]: Why do you catch here CastClassException? It's a bit strange for me especially that it actually informs that casting to HttpURLConnection failed, not that the downloaded file does not exist...

**Review by helinek — COMMENTED** [review]
Please provide the response to the comment

File: src/main/java/pl/psnc/dei/iiif/Converter.java (hunk 1 above)
Comment (kacperwleklak) [reply src/main/java/pl/psnc/dei/iiif/Converter.java]: Refactored this to be more readable

**Pull request merged**